ethers-solc/src/artifacts/mod.rs

```diff
@@ -159,7 +159,11 @@ impl CompilerInput {
         if self.is_yul() {
             warn!("omitting remappings supplied for the yul sources");
         } else {
-            self.settings.remappings = remappings;
+            if self.settings.remappings.is_empty() {
+                self.settings.remappings = remappings;
+            } else {
+                warn!("remappings in settings conflicts with path configurations");
+            }
```

Comment:
I disagree here, 
I'd expect that a function called `with_remappings` would always set the remappings

Reply:
That's the point where I'm open to discussion.

Another way is to modify here: https://github.com/gakonst/ethers-rs/blob/594627dc1c3b490ba8f513f8f5e23d11448cbcf8/ethers-solc/src/compile/project.rs#L546 by checking `Settings` there. Are you fine with that?